socorro/unittest/cron/jobs/test_archivescraper.py

```diff
@@ -0,0 +1,313 @@
+
+    :arg path: a url path
+
+    :returns: iterator over all the (parent path, child link) pairs
+
+    For example:
+
+    >>> list(absolute_links_parents('/pub/firefox/candidates/64.0b13-candidates/'))
+    [('/', 'pub/'), ('/pub/', 'firefox/'), ('/pub/firefox/', 'candidates/'),
+    ('/pub/firefox/candidates/', '64.0b13-candidates/')]
+
+    """
+    path_parts = path.strip('/').split('/')
+    for i, path_part in enumerate(path_parts):
+        parent_path = '/' + '/'.join(path_parts[0:i])
+        if parent_path != '/':
+            parent_path += '/'
+
+        if not path_part.endswith('.json'):
+            path_part += '/'
+
+        yield (parent_path, path_part)
+
+
+def render_index_template(path, links):
+    """Given a list of links, generates an HTML index page"""
+    jinja_env = Environment()
+    index_template = jinja_env.from_string(INDEX_PAGE_TEMPLATE)
+    return index_template.render(path=path, links=links)
+
+
+def setup_site(req_mock, root, file_contents):
+    """Generates all the req_mock.get calls to simulate a site
+
+    :arg req_mock: a requests_mock mock instance
+    :arg str root: the root path to prepend to all paths
+    :arg list file_contents: list of (path, contents)
+
+    """
+    root = root.rstrip('/')
+
+    # map of path -> set of links
+    path_to_links = {}
+
+    for path, contents in file_contents:
+        path = root + '/' + path
+        req_mock.get(HOST + path, json=contents)
+
+        for parent_path, link in absolute_links_parents(path):
+            path_to_links.setdefault(parent_path, set()).add(link)
+
+    for path, links in path_to_links.items():
+        links = [
+            {
+                'path': path + link,
+                'name': link
+            }
+            for link in links
+        ]
+
+        index_page = render_index_template(path, links)
+        req_mock.get(HOST + path, text=index_page)
+
+
+@pytest.fixture
+def config():
+    """Pytest fixture that returns config for ArchiveScraperCronApp"""
+    # Configuration for crontabber jobs is buried in CrontabberApp, so we
+    # create one of those and then extract the configuration that we want. It's
+    # either that or we have to hard-code expanding all the configuration bits.
+    config_manager = get_config_manager(
+        jobs='socorro.cron.jobs.archivescraper.ArchiveScraperCronApp|1h',
+        overrides={
+            'crontabber.class-ArchiveScraperCronApp.base_url': HOST + '/pub/',
+        }
+    )
+    with config_manager.context() as config:
+        crontabberapp = CronTabberApp(config)
+        class_config = crontabberapp.config.crontabber['class-ArchiveScraperCronApp']
+        yield class_config
+
+
+@pytest.fixture
+def req_mock():
+    with requests_mock.mock() as mock:
+        yield mock
+
+
+class TestArchiveScraper(object):
+    @classmethod
+    def setup_class(cls):
+        cls.conn = psycopg2.connect(os.environ['DATABASE_URL'])
```

Comment:
This is probably not a great idea if we ever do parallel execution of tests, but it's fine for now.